libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/destination/DestinationAction.kt

```diff
@@ -1,7 +1,14 @@
 package com.mapbox.navigation.dropin.component.destination
 
+import com.mapbox.api.geocoding.v5.models.CarmenFeature
+import com.mapbox.geojson.Point
 import com.mapbox.navigation.dropin.model.Destination
 
 sealed class DestinationAction {
     data class SetDestination(val destination: Destination?) : DestinationAction()
+
+    data class DidReverseGeocode(
```

Comment:
Curious on the data values. It seems each action is responsible for carrying some of the state. the `Destination` object has both a `Point` and `List<CarmenFeature>`. `DidReverseGeocode` includes both.

Seems like it could look like this 🤔 



``` kotlin
data class SetDestination(val destination: Point) : DestinationAction()
data class DidReverseGeocode(val features: List<CarmenFeature>) : DestinationAction()

internal data class DestinationState(
    val destination: Point? = null
    val features: List<CarmenFeature>? = null
)
```

Reply:
Since `features` are for the specific `Point` they should be part of the `Destination`. Reverse Geocode is an asynchronous operations. `DidReverseGeocode` might be fired after destination already got changed. Including both `features` and `point` in `DidReverseGeocode` action allows `DestinationViewModel` [perform check](https://github.com/mapbox/mapbox-navigation-android/pull/5571/files#diff-dce1a679385bfa22cfe4e1de42d789ca9806d7a26564bd6f23185047c82c3bafR22) before mutating its state.

Reply:
ah right, makes sense for `DidReverseGeocode` 👍 . 

I don't see where the features are used in the `SetDestination` action, so not sure why we need it to share the `Destination` object. `DestinationState` can be decoupled from the action parameters

```
LongPressMapComponent { context.dispatch(DestinationAction.SetDestination(Destination(point)))
BackPressManager {      context.dispatch(DestinationAction.SetDestination(null))
```